src/bogo_shim/config.json

```diff
@@ -40,33 +41,10 @@
         "TooManyChangeCipherSpec-Client-TLS13": "Limits on the number of CCS are not implemented",
         "TooManyKeyUpdates": "Limits on the number of KeyUpdates are not implemented",
 
-        "CertificateVerificationSucceed-Client-TLS13-*" : "TLS 1.3 session resumption is NYI",
-        "Client-VerifyDefault-*-TLS13" : "TLS 1.3 session resumption is NYI",
-        "Client-Verify-*-TLS13" : "TLS 1.3 session resumption is NYI",
-        "ExportKeyingMaterial-TLS13" : "TLS 1.3 session resumption is NYI",
-        "InvalidPSKIdentity-TLS13" : "TLS 1.3 session resumption is NYI",
-        "NegotiatePSKResumption-TLS13" : "TLS 1.3 session resumption is NYI",
-        "Resume-Client-CipherMismatch-TLS13" : "TLS 1.3 session resumption is NYI",
-        "Resume-Client-Mismatch-TLS13-TLS12-TLS" : "TLS 1.3 session resumption is NYI",
-        "Resume-Client-TLS13-TLS13-TLS" : "TLS 1.3 session resumption is NYI",
-        "TLS-TLS13-AES_128_GCM_SHA256-client" : "TLS 1.3 session resumption is NYI",
-        "TLS-TLS13-AES_256_GCM_SHA384-client" : "TLS 1.3 session resumption is NYI",
-        "TLS-TLS13-CHACHA20_POLY1305_SHA256-client" : "TLS 1.3 session resumption is NYI",
-        "TLS12SessionID-TLS13" : "TLS 1.3 session resumption is NYI",
-        "TLS13-HonorServerSessionTicketLifetime" : "TLS 1.3 session resumption is NYI",
-        "TLS13-TestBadTicketAge-Client" : "TLS 1.3 session resumption is NYI",
-        "TLS13-TestValidTicketAge-Client" : "TLS 1.3 session resumption is NYI",
-        "TLS13SessionID-TLS13" : "TLS 1.3 session resumption is NYI",
-        "TolerateServerNameAck-TLS-TLS13" : "TLS 1.3 session resumption is NYI",
-        "OCSPStapling-Client-TLS13-*" : "TLS 1.3 session resumption is NYI",
-        "Resume-Client-NoResume-TLS12-TLS13-TLS": "TLS 1.3 session resumption is NYI",
-        "Resume-Client-Mismatch-TLS12-TLS13-TLS": "TLS 1.3 session resumption is NYI",
-        "Ticket-Forbidden-TLS13": "TLS 1.3 session resumption is NYI",
-        "Resume-Client-PRFMismatch-TLS13": "TLS 1.3 session resumption is NYI",
-        "TLS13-HelloRetryRequest-Client-*": "TLS 1.3 session resumption is NYI",
-        "TLS13-TicketAgeSkew-*": "TLS 1.3 session resumption is NYI",
-        "CurveID-Resume-Client-TLS13": "TLS 1.3 session resumption is NYI",
-        "ALPNClient-TLS-TLS13": "TLS 1.3 session resumption is NYI",
+        "TLS12SessionID-TLS13": "We don't offer TLS 1.3 when a TLS 1.2 session was found",
+        "Ticket-Forbidden-TLS13": "We don't offer TLS 1.3 when a TLS 1.2 session was found",
+        "Resume-Client-NoResume-TLS12-TLS13-TLS": "We don't offer TLS 1.3 when a TLS 1.2 session was found",
+        "Resume-Client-Mismatch-TLS12-TLS13-TLS": "We don't offer TLS 1.3 when a TLS 1.2 session was found",
```

Comment:
We basically query the `Session_Manager` before instantiating the TLS 1.2 or TLS 1.3 implementation. If we find a TLS 1.2 session we don't even bother offering TLS 1.3 in the first place. Hence, we cannot fulfil those tests. Though, I think it's worth discussing whether the described behaviour is actually wise.

Reply:
Would of course be nice to switch to TLS 1.3. even than an older session exist. But unless you think the new some API change for that I don't think it is important to have it in the first version.